src/movies.js

```diff
@@ -1,22 +1,60 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+function getAllDirectors(moviesArray) {
+    return moviesArray.map(movies => movies.director);
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(moviesArray) {
+    return moviesArray.filter(movies => movies.genre.includes('Drama') && movies.director === 'Steven Spielberg').length;
+}
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
+function scoresAverage(moviesArray) {
+    if (moviesArray.length === 0) return 0;
+    else {
+        const validMovies = [...moviesArray];
```

Comment:
The `.reduce()` method already returns a new array, so you don't need to copy before it
```suggestion
```